django_cloud_deploy/crash_handling/__init__.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A module handles crashes of the tool."""
+
+import os
+import platform
+import subprocess
+import sys
+import traceback
+import urllib.parse
+import webbrowser
+
+import jinja2
+
+from django_cloud_deploy.cli import io
+from django_cloud_deploy import __version__
+
+# TODO: Add an issue label for issues reported by users
+_REQUEST_URL_TEMPLATE = (
+    ('https://github.com/GoogleCloudPlatform/django-cloud-deploy/issues/new?'
+     'body={}&title={}'))
+
+
+with open(os.path.join(os.path.dirname(__file__), 'template',
+                       'issue_template.txt')) as f:
+    _ISSUE_TEMPLATE = f.read()
+
+
+def handle_crash(err: Exception, command: str,
+                 console: io.IO = io.ConsoleIO()):
+    """The tool's crashing handler.
+
+    Args:
+        err: The exception that was raised.
+        command: The command causing the exception to get thrown,
+            e.g. 'django-cloud-deploy new'.
+        console: Object to use for user I/O.
+    """
+
+    # TODO: Only handle crashes caused by our code, not user's code.
+    # When deploying, our tool will run the code of user's Django project.
+    # If user's code has a bug, then the traceback will contain the directory
+    # path of their Django project. We will put traceback in a public issue,
+    # but user's might not want to put these kind of information to public,
+    # so we plan to only handle crashes caused by our code.
+    console.tell(
+        ('Your "{}" failed due to an internal error - please try again.'
+         '\n\n'
+         'You can report this error by filing a bug on Github. If you agree,\n'
+         'a browser window will open and an Github issue will be\n'
+         'pre-populated with the details of this crash.\n').format(command))
+    while True:
+        ans = console.ask('Would you like to file a bug? [y/N]: ')
+        ans = ans.strip().lower()
+        if ans in ['y', 'n']:
```

Comment:
I think that you could break on the empty string too since 'N' is supposed to be a default.

Reply:
Done.